src/gamestate/game_scene.cpp

```diff
@@ -464,8 +464,11 @@ void on_lbutton_up(sys::state& state, int32_t x, int32_t y, sys::key_modifiers m
 	if(state.user_settings.left_mouse_click_hold_and_release && state.ui_state.left_mouse_hold_target) {
 		on_lbutton_up_ui_click_hold_and_release(state, x, y, mod);
 	}
-	on_lbutton_up_map(state, x, y, mod);
-	state.current_scene.lbutton_up(state);
+
+	if(!state.ui_state.under_mouse) {
+		on_lbutton_up_map(state, x, y, mod);
```

Comment:
I believe the map lbutton up belongs outside this check because, if you were dragging a selection box for units, you probably wouldn't want the selection process to continue just because you happened to end over a ui element